api/gateway-operator/v1beta1/controlplane_conversion.go

```diff
@@ -67,25 +68,35 @@ func (c *ControlPlane) ConvertTo(dstRaw conversion.Hub) error {
 
 	dst.ObjectMeta = c.ObjectMeta
 
+	hasGatewayAsOwner := lo.ContainsBy(c.GetOwnerReferences(), func(owner metav1.OwnerReference) bool {
+		return strings.HasPrefix(owner.APIVersion, gatewayv1.GroupName) &&
+			owner.Kind == "Gateway"
+	})
+
 	// Setting IngressClass wasn't required in v1beta1, but for v2beta1 it is,
+	// when ControlPlane is not managed by a Gateway,
 	// thus fill with a default value to make it work without hassle.
-	class := lo.FromPtrOr(c.Spec.IngressClass, "kong")
-	if err := c.Spec.convertTo(&dst.Spec.ControlPlaneOptions, &class); err != nil {
+	class := c.Spec.IngressClass
+	if !hasGatewayAsOwner && class == nil {
+		class = lo.ToPtr("kong")
```

Comment:
nit: [Here we](https://github.com/Kong/kubernetes-configuration/blob/3e32e9d6e7e5a1de92eb5b06dd3f89b7d6060bdb/api/gateway-operator/v1beta1/controlplane_conversion.go#L72) also set a "default" class `kong`, maybe it's worth pulling `kong` to some shared constant?

Reply:
It's the same file isn't it? Searching for `"kong"` yields the same result it seems, doesn't it?